main.tf

```diff
@@ -1,4 +1,40 @@
-resource "google_storage_bucket" "example" {
-  name     = var.name
-  location = "EU"
+data "aws_caller_identity" "current" {}
+
+# Create a secret in AWS Secrets Manager to store the upstream registry name.
+resource "aws_secretsmanager_secret" "ecr_pullthroughcache" {
+  name = "ecr-pullthroughcache/${var.upstream_registry}"
+}
+
+resource "aws_secretsmanager_secret_version" "ecr_pullthroughcache" {
+  secret_id     = aws_secretsmanager_secret.ecr_pullthroughcache.id
+  secret_string = jsonencode(var.credentials)
+
+  lifecycle {
+    ignore_changes = [secret_string]
+  }
+}
+
+resource "aws_ecr_pull_through_cache_rule" "ecr_pullthroughcache" {
+  ecr_repository_prefix = var.upstream_registry
+  upstream_registry_url = var.upstream_registry_url
+  credential_arn        = aws_secretsmanager_secret.ecr_pullthroughcache.arn
+}
+
+resource "aws_iam_policy" "ecr_pullthroughcache" {
+  name        = "ecr-pullthroughcache"
```

Comment:
Put something in the name related to the specific pull through cache targeted by this policy. If a project will installs this module twice to proxy two different repos, this will fail.